src/Microsoft.AspNetCore.Razor.VSCode/src/Diagnostics/ReportIssueCreator.ts

```diff
@@ -0,0 +1,244 @@
+        if (csharpTextDocument) {
+            csharpContent = `${csharpContent}
+${csharpTextDocument.getText()}`;
+        } else {
+            csharpContent = `${csharpContent}
+Unable to resolve VSCode's version of CSharp`;
+        }
+
+        return csharpContent;
+    }
+
+    // Protected for testing
+    protected async getProjectedHtml(razorDocument: IRazorDocument) {
+        let htmlContent = `////////////////////// Projected Html as seen by extension ///////////////////////
+${razorDocument.htmlDocument.getContent()}
+
+
+////////////////////// Projected Html as seen by VSCode ///////////////////////`;
+
+        const htmlTextDocument = await this.vscodeApi.workspace.openTextDocument(razorDocument.htmlDocument.uri);
+        if (htmlTextDocument) {
+            htmlContent = `${htmlContent}
+${htmlTextDocument.getText()}`;
+        } else {
+            htmlContent = `${htmlContent}
+Unable to resolve VSCode's version of Html`;
+        }
+
+        return htmlContent;
+    }
+
+    // Protected for testing
+    protected getExtensionVersion(): string {
+        const extension = this.vscodeApi.extensions.getExtension(razorExtensionId);
+        if (!extension) {
+            return 'Unable to find Razor extension version.';
+        }
+        return extension.packageJSON.version;
+    }
+
+    // Protected for testing
+    protected getInstalledExtensions() {
+        const extensions: Array<vscode.Extension<any>> = this.vscodeApi.extensions.all
+            .filter(extension => extension.packageJSON.isBuiltin === false);
+
+        return extensions.sort(this.sortExtensions);
+    }
+
+    // Protected for testing
+    protected sortExtensions(a: vscode.Extension<any>, b: vscode.Extension<any>): number {
+        const aName = a.packageJSON.name.toLowerCase();
+        const bName = b.packageJSON.name.toLowerCase();
+        if (aName < bName) {
+            return -1;
+        }
+        if (aName > bName) {
+            return 1;
+        }
+        return 0;
+    }
+
+    // Protected for testing
+    protected generateExtensionTable() {
+        const extensions = this.getInstalledExtensions();
+        if (extensions.length <= 0) {
+            return 'none';
+        }
+
+        const tableHeader = `|Extension|Author|Version|\n|---|---|---|`;
+        const table = extensions.map(
+            (e) => `|${e.packageJSON.name}|${e.packageJSON.publisher}|${e.packageJSON.version}|`).join(os.EOL);
+
+        const extensionTable = `
+${tableHeader}\n${table};
+`;
+
+        return extensionTable;
+    }
+
+    private getDotnetInfo(): Promise<string> {
+        return new Promise<string>((resolve) => {
+            try {
+                cp.exec('dotnet --info', { cwd: process.cwd(), maxBuffer: 500 * 1024 }, (error, stdout, stderr) => {
+                    if (error) {
+                        throw error;
+                    } else if (stderr && stderr.length > 0) {
+                        throw new Error(stderr);
```

Comment:
I don't think we should `throw` either here or on L232. Nothing will catch the async exception. I think we should use `reject` instead (also changing L228 to `return new Promise<string>((resolve, reject) => {`).

Reply:
Interesting, I figured invoking cp.exec in the promise would throw from the inside but I see what you're saying. Good call. I'll fix.